ipam/azure_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package ipam
+
+import (
+	"encoding/json"
+	"github.com/Azure/azure-container-networking/common"
+	cniTypesCurr "github.com/containernetworking/cni/pkg/types/current"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+)
+
+var ipamQueryUrl = "localhost:42424"
+var ipamQueryResponse = "" +
+	"<Interfaces>" +
+	"	<Interface MacAddress=\"*\" IsPrimary=\"true\">" +
+	"		<IPSubnet Prefix=\"10.0.0.0/16\">" +
+	"			<IPAddress Address=\"10.0.0.4\" IsPrimary=\"true\"/>" +
+	"			<IPAddress Address=\"10.0.0.5\" IsPrimary=\"false\"/>" +
+	"		</IPSubnet>" +
+	"		<IPSubnet Prefix=\"10.1.0.0/16\">" +
+	"			<IPAddress Address=\"10.1.0.4\" IsPrimary=\"false\"/>" +
+	"		</IPSubnet>" +
+	"	</Interface>" +
+	"</Interfaces>"
+
+// Handles queries from IPAM source.
+func handleIpamQuery(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(ipamQueryResponse))
+}
+
+func parseResult(stdinData []byte) (*cniTypesCurr.Result, error) {
+	result := &cniTypesCurr.Result{}
+	err := json.Unmarshal(stdinData, result)
```

Comment:
current style is to merge with if statement for single returns
e.g.

if err := json.Marshal...; err != nil {
    ...
}

Reply:
OK, thanks!